app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -269,34 +284,84 @@ open class ProfilesViewModel(
     _tempProfilePictureBitmap.value = bitmap
   }
 
+  /**
+   * Saves the edited profile to the state.
+   *
+   * @param profile The profile to be saved.
+   */
+  fun saveEditedProfile(profile: Profile) {
+    _editedCurrentProfile.value = profile
+  }
+
   /** Clears the temporary profile picture Bitmap. */
   fun clearTempProfilePictureBitmap() {
     _tempProfilePictureBitmap.value = null
   }
 
+  /** Clears the edited profile from the state. */
+  private fun clearEditedProfile() {
+    _editedCurrentProfile.value = null
+  }
+
+  /** Resets every state related to profile edition. */
+  fun resetProfileEditionState() {
+    clearTempProfilePictureBitmap()
+    clearEditedProfile()
+    _pictureChangeState.value = ProfilePictureState.UNCHANGED
+  }
+
   /**
    * Processes the cropped image by compressing it if it is too large.
    *
    * @param bitmap The cropped Bitmap to be processed.
    */
   fun processCroppedImage(bitmap: Bitmap) {
     _tempProfilePictureBitmap.value = compressImageIfTooLarge(bitmap)
+    _pictureChangeState.value = ProfilePictureState.TO_UPLOAD
+  }
+
+  fun validateProfileChanges(context: Context) {
+    when (_pictureChangeState.value) {
+      ProfilePictureState.TO_UPLOAD ->
+          validateAndUploadProfilePicture(context) { url ->
+            val newProfile = _editedCurrentProfile.value!!.copy(profilePictureUrl = url)
```

Comment:
Small detail / Question : If the function "validateProfileChanges(...)" is called before "saveEditedProfile" at the beginning, then "_editedCurrentProfile.value" can be null, so is it safe to set "!!" here ?

Reply:
I understand your concern. There is only one place this function is called, and I used saveEditedProfile before. Now that I think of it I could have passed the changes as parameter and avoid this saving call from `ProfileEdit`, but as I am now well into test development, I don't want to break everything